pkg/stack_mgr/stack_service.go

```diff
@@ -164,6 +186,33 @@ func (s *StackService) Remove(ctx context.Context, stackName string) error {
 	return nil
 }
 
+func (s *StackService) AddWithLock(ctx context.Context, stackName string) (*Stack, error) {
+	distributedLock, err := s.getDistributedLock()
+	if err != nil {
+		return nil, err
+	}
+	defer distributedLock.Close(ctx)
+
+	lock, err := distributedLock.AcquireLock(ctx, s.writePath)
+	if err != nil {
+		return nil, err
+	}
+
+	stack, err := s.Add(ctx, stackName)
+
+	_, err = distributedLock.ReleaseLock(ctx, lock)
+	if err != nil {
+		return nil, err
+	}
+
+	return stack, err
+}
+
+func (s *StackService) getDistributedLock() (*backend.DistributedLock, error) {
+	lockConfig := backend.DistributedLockConfig{DynamodbTableName: s.backend.Conf().DynamoLocktableName}
```

Comment:
Do we expect this lock to need a longer lease duration than the default of 10 seconds?

Reply:
Testing showed this was taking ~1.5 seconds for removing a stack and ~49s for adding a stack. I've refactored to reduce the time that a lock must be held for to keep it under a couple seconds